boot/seal.go

```diff
@@ -101,16 +101,17 @@ func sealKeyToModeenv(key secboot.EncryptionKey, model *asserts.Model, modeenv *
 		return fmt.Errorf("cannot prepare for key sealing: %v", err)
 	}
 	// make sure relevant locations exist
-	for _, p := range []string{InitramfsEncryptionKeyDir, InstallHostFDEDataDir} {
+	for _, p := range []string{InitramfsEncryptionKeyDir, InstallHostFDEDataDir, InstallHostFDESaveDir} {
+		// XXX: should that be 0700 ?
 		if err := os.MkdirAll(p, 0755); err != nil {
 			return err
 		}
 	}
 	sealKeyParams := &secboot.SealKeyParams{
 		ModelParams:          modelParams,
 		KeyFile:              filepath.Join(InitramfsEncryptionKeyDir, "ubuntu-data.sealed-key"),
-		TPMPolicyAuthKeyFile: filepath.Join(InstallHostFDEDataDir, "tpm-policy-auth-key"),
-		TPMLockoutAuthFile:   filepath.Join(InstallHostFDEDataDir, "tpm-lockout-auth"),
+		TPMPolicyAuthKeyFile: filepath.Join(InstallHostFDESaveDir, "tpm-policy-auth-key"),
```

Comment:
This is the new writable, so once we bind mount /run/mnt/ubuntu over /var/lib/snapd/save in run mode, the files will become inaccessible?

Reply:
mmh, if ubuntu-save exists we want to use it mounted, I think gadget/install mounts everything? so this is wrong and I misread the code?

Reply:
iirc, yes the gadget/install code from install.Run() will mount all the partitions at `/run/mnt/<partition-name>` so during install mode `/run/mnt/ubuntu-save` exists and refers to the actual partition for `ubuntu-save`.